test/test_jit_fuser_te.py

```diff
@@ -1803,5 +1807,130 @@ def eager(x, y, z: int):
+            'nn.functional.hardswish',
+            'nn.functional.hardtanh',
+            'pow',
+            'reciprocal',
+            'round',
+            'rsqrt',
+            'sigmoid',
+            'sin',
+            'sinh',
+            'sqrt',
+            'sub',
+            'tan',
+            'tanh',
+            'trunc',
+            'unsqueeze',
+        ]
+        # These are ops that don't work (or are incomplete) and need to be fixed
+        fail_ops = ['matmul', 'permute', 'frac', '__rmatmul__']
+        try:
+            sample_inputs_itr = op.sample_inputs(device, dtype, requires_grad=False)
+            for sample_input in sample_inputs_itr:
+                arg_values = [sample_input.input] + list(sample_input.args)
+                kwarg_values = sample_input.kwargs
+                param_names = []
+                param_values = []
+                fx_args = []
+                for idx, v in enumerate(arg_values):
+                    if isinstance(v, torch.Tensor):
+                        param_names.append(f"arg_{idx}")
+                        param_values.append(v)
+                        fx_args.append(param_names[-1])
+                    else:
+                        fx_args.append(f'{repr(v)}')
+
+                for k, v in kwarg_values.items():
+                    if isinstance(v, torch.Tensor):
+                        param_names.append(k)
+                        param_values.append(v)
+                        fx_args.append(f'{k} = {k}')
+                    else:
+                        fx_args.append(f'{k} = {repr(v)}')
+
+                code = f"""
+def f({', '.join(param_names)}):
+    return op.op({', '.join(fx_args)})"""
+                g = {'torch': torch, 'inf' : math.inf, 'op': op}
+                exec(code, g)
+                f = g['f']
+                f.__module__ = 'test'
+                out = f(*param_values)
+                # NNC currently doesn't support lowering ops with more than one
+                # output
+                if isinstance(out, tuple):
+                    continue
+
+                # NNC currently oftens segfault when asked to lower ops with 0-dim tensor outputs
+                if isinstance(out, torch.Tensor) and out.dim() == 0:
+                    continue
+                ts_g = torch.jit.trace(f, param_values)
+                kernel = torch._C._te.TensorExprKernel(ts_g.graph)
+                self.assertEqual(kernel.run(tuple(param_values)), f(*param_values))
+                self.assertEqual(kernel.fallback(tuple(param_values)), f(*param_values))
+
+        except Exception as e:
+            self.assertTrue(op.name not in work_list)
+            if "Unhandled node kind" in str(e):
+                print(e)
+                return
+            if "UNSUPPORTED DTYPE" in str(e):
+                print(e)
+                return
+            if "UNSUPPORTED DTYPE" in str(e):
+                print(e)
+                return
```

Comment:
Should we rethrow the exception in this case? Also, the dtype check is duplicated

Reply:
Well, if we rethrow the exception the test will fail :thinking:

Reply:
I mean, fail is the correct behavior here :) Could you just comment out the ops that are failing from the list and put a TODO to fix them?

Reply:
The reason it's written this way is that the vast majority of ops are actually not supported (i.e. they don't have lowerings). So, the ops I have in here are a whitelist.

I wouldn't be too opposed to having a very large blacklist, but one issue I've found with OpInfo is that it's an open database - that is, other people will add tests. So, keeping a blacklist causes a maintenance burden whenever people add new tests that fail.

If I could, I would prefer something like "expectedFailure", but I don't think there's any way of doing that here.

Reply:
So, from the ops from the work_list there are some that throw exceptions here, right? Can we comment them out from the work_list until we fix them?

Reply:
No there aren't - the "work_list" is only ops that work without exceptions.